frontends/article/article-page.tsx

```diff
@@ -5,67 +5,60 @@ import Image from 'next/image';
 import { useAppDispatch } from 'store';
 import { useLazyCheckIfLikedDislikedQuery, useLikeDislikeMutation } from 'store/apis';
-import { openLoginModal, toggleCommentsSidebar } from 'store/states';
+import { openLoginModal } from 'store/states';
 import { addUriToImageBlocks, toDateString } from 'utils';
-import { UPPER_BLUE_COLOR } from 'variables';
-import { ICONS } from 'variables/icons';
+import { ICONS, UPPER_BLUE_COLOR } from 'variables';
 
 import styles from './article.module.scss';
 import { IArticleProps } from './article.types';
-import { ArticleActions } from './components';
-
-const LikeIcon = ICONS.like;
-const DislikeIcon = ICONS.dislike;
-const CommentIcon = ICONS.comment;
+import { ArticleActionIcons, ArticleActions } from './components';
 
 const toUzbDateString = (date: Date | string): string => toDateString(date, { month: 'short' });
+const LikeIcon = ICONS.like;
 
-export const Article: FC<IArticleProps> = (props) => {
-  const { viewCount, publishedDate, updatedDate, blocks, id, likeCount, dislikeCount } = props;
-  const [editorInstance, setEditorInstance] = useState<EditorJS | null>(null);
+export const Article: FC<IArticleProps> = (article) => {
+  const {
+    id,
+    viewCount = 0,
+    publishedDate,
+    updatedDate,
+    blocks = [],
+    likeCount = 0,
+    dislikeCount = 0,
+  } = article || {};
   const [likeDislikeCount, setLikeDislikeCount] = useState<number>(likeCount - dislikeCount);
+  const [editorInstance, setEditorInstance] = useState<EditorJS | null>(null);
+  const [isSharePopupOpen, setIsSharePopupOpen] = useState<boolean>(false);
+  const [likeDislikeArticle, likeDislikeRes] = useLikeDislikeMutation();
   const { isAuthenticated } = useAuth();
-  const { themeColors } = useTheme();
   const dispatch = useAppDispatch();
-  const [likeDislikeArticle, likeDislikeRes] = useLikeDislikeMutation();
   const [checkIfLikedDislikedQuery, { data: isLikedOrDisliked }] =
     useLazyCheckIfLikedDislikedQuery();
+  const { themeColors } = useTheme();
 
   const likeDislike = (value: -1 | 1): void => {
     if (!isAuthenticated) {
       dispatch(openLoginModal());
       return;
     }
-    if (likeDislikeRes.isLoading || value === isLikedOrDisliked) return;
+    if (likeDislikeRes.isLoading || value === isLikedOrDisliked || !id) return;
     likeDislikeArticle({ id, value }).then(() => {
       setLikeDislikeCount((prev) => prev + value - (isLikedOrDisliked || 0));
     });
   };
 
-  const commentIconClickHandler = (): void => {
-    dispatch(toggleCommentsSidebar());
-  };
-
-  useEffect(() => {
-    if (editorInstance?.isReady) {
-      const main = document.querySelector('#main');
-      main?.scrollIntoView({ block: 'start' });
-      main?.scrollTo(0, 0);
-    }
-  }, [editorInstance?.isReady]);
-
-  useEffect(() => {
-    if (isAuthenticated) {
-      checkIfLikedDislikedQuery(id);
-      setLikeDislikeCount(likeCount - dislikeCount);
+  const likeIcon = useMemo((): JSX.Element => {
```

Comment:
this logic should be inside article-actions-icons.tsx component and rename the file to article-action-icons.tsx